lib/commands/general.js

```diff
@@ -27,21 +27,46 @@ commands.doSendKeys = async function (params) {
- *                        `iso-8601`format by default like `2018-06-09T16:21:54+0900`.
- *                        e.g.: `'+%Y-%m-%dT%T%z'`
+ * @param {String} format A optional format for `adb shell date`.
+ *                        `iso-8601`format is default like `2018-06-09T16:21:54+0900` by `'+%Y-%m-%dT%T%z'`
+ *
+ *                        usage: date [-u] [-r FILE] [-d DATE] [+DISPLAY_FORMAT] [-D SET_FORMAT] [SET]
+ *                        Set/get the current date/time. With no SET shows the current date.
+ *
+ *                        Default SET format is "MMDDhhmm[[CC]YY][.ss]", that's (2 digits each)
+ *                        month, day, hour (0-23), and minute. Optionally century, year, and second.
+ *                        Also accepts "@UNIXTIME[.FRACTION]" as seconds since midnight Jan 1 1970.
+ *
+ *                        -d	Show DATE instead of current time (convert date format)
+ *                        -D	+FORMAT for SET or -d (instead of MMDDhhmm[[CC]YY][.ss])
+ *                        -r	Use modification time of FILE instead of current date
+ *                        -u	Use UTC instead of current timezone
+ *
+ *                        +FORMAT specifies display format string using these escapes:
+ *
+ *                        %% literal %             %n newline              %t tab
+ *                        %S seconds (00-60)       %M minute (00-59)       %m month (01-12)
+ *                        %H hour (0-23)           %I hour (01-12)         %p AM/PM
+ *                        %y short year (00-99)    %Y year                 %C century
+ *                        %a short weekday name    %A weekday name         %u day of week (1-7, 1=mon)
+ *                        %b short month name      %B month name           %Z timezone name
+ *                        %j day of year (001-366) %d day of month (01-31) %e day of month ( 1-31)
+ *                        %s seconds past the Epoch
+ *
+ *                        %U Week of year (0-53 start sunday)   %W Week of year (0-53 start monday)
+ *                        %V Week of year (1-53 start monday, week < 4 days not part of this year)
+ *
+ *                        %D = "%m/%d/%y"    %r = "%I : %M : %S %p"   %T = "%H:%M:%S"   %h = "%b"
+ *                        %x locale date     %X locale time           %c locale date/time
+ *
  * @return {String} Return date formatted by `format`. `+%Y-%m-%dT%T%z` is by default.
  * @throws {Error} If the adb command doesn't return device date, error throes
  */
 commands.getDeviceTime = async function (format) {
   log.info('Attempting to capture android device date and time');
   try {
-    let formatString = '+%Y-%m-%dT%T%z';
-    if (_.isString(format)) {
-      formatString = format;
-    }
-    const out = await this.adb.shell(['date', formatString]);
-    return out.trim();
+    const formatString = _.isString(format) ? format : '+%Y-%m-%dT%T%z';
```

Comment:
It might be better to extract this assignment from try/catch block

Reply:
Also I was thinking about assigning the default argument value directly to the argument like 
`commands.getDeviceTime = async function (format = '+%Y-%m-%dT%T%z') {`

Reply:
oh, JS has the default argument... thanks